components/drawer/index.tsx

```diff
@@ -61,271 +67,269 @@ interface InternalDrawerProps extends DrawerProps {
+      maskClosable = true,
+      mask = true,
+      level = null,
+      keyboard = true,
+      push = defaultPushState,
+      closeIcon = <CloseOutlined />,
+      bodyStyle,
+      drawerStyle,
+      prefixCls,
+      className,
+      direction,
+      visible,
+      children,
+      zIndex,
+      destroyOnClose,
+      style,
+      title,
+      headerStyle,
+      onClose,
+      footer,
+      footerStyle,
+      ...rest
+    },
+    ref,
+  ) => {
+    const forceUpdate = useForceUpdate();
+    const [internalPush, setPush] = React.useState(false);
+    const parentDrawer = React.useRef<DrawerRef | null>(null);
+    const destroyClose = React.useRef<boolean>(false);
+
+    React.useEffect(() => {
+      // fix: delete drawer in child and re-render, no push started.
+      // <Drawer>{show && <Drawer />}</Drawer>
+      if (visible && parentDrawer.current) {
+        parentDrawer.current.push();
       }
-    }
-  }
 
-  public componentWillUnmount() {
-    // unmount drawer in child, clear push.
-    if (this.parentDrawer) {
-      this.parentDrawer.pull();
-      this.parentDrawer = null;
-    }
-  }
-
-  push = () => {
-    if (this.props.push) {
-      this.setState({ push: true });
-    }
-  };
-
-  pull = () => {
-    if (this.props.push) {
-      this.setState({ push: false });
-    }
-  };
+      return () => {
+        if (parentDrawer.current) {
+          parentDrawer.current.pull();
+          parentDrawer.current = null;
+        }
+      };
+    }, []);
+
+    React.useEffect(() => {
+      if (parentDrawer.current) {
+        if (visible) {
+          parentDrawer.current.push();
+        } else {
+          parentDrawer.current.pull();
+        }
+      }
+    }, [visible]);
 
-  onDestroyTransitionEnd = () => {
-    const isDestroyOnClose = this.getDestroyOnClose();
-    if (!isDestroyOnClose) {
-      return;
-    }
-    if (!this.props.visible) {
-      this.destroyClose = true;
-      this.forceUpdate();
-    }
-  };
+    const operations = React.useMemo(
+      () => ({
+        push() {
+          if (push) {
+            setPush(true);
+          }
+        },
+        pull() {
+          if (push) {
+            setPush(false);
+          }
+        },
+      }),
+      [],
+    );
 
-  getDestroyOnClose = () => this.props.destroyOnClose && !this.props.visible;
+    React.useImperativeHandle(ref, () => operations, [operations]);
 
-  getPushDistance = () => {
-    const { push } = this.props;
-    let distance: number | string;
-    if (typeof push === 'boolean') {
-      distance = push ? defaultPushState.distance : 0;
-    } else {
-      distance = push!.distance;
-    }
-    return parseFloat(String(distance || 0));
-  };
+    const isDestroyOnClose = destroyOnClose && !visible;
 
-  // get drawer push width or height
-  getPushTransform = (placement?: placementType) => {
-    const distance = this.getPushDistance();
+    const onDestroyTransitionEnd = React.useCallback(() => {
```

Comment:
Why use `useCallback`?

Reply:
That was a mistake, I resolved that.
thanks